File: src/main/java/io/github/saneea/dvh/AppExitException.java
package io.github.saneea.dvh;

public class AppExitException extends RuntimeException {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	public interface ExitCode {
		int OK = 0;
		int ERROR = 1;
	}

	private final int exitCode;

	public AppExitException(int exitCode) {
		this(exitCode, null);
	}

	public AppExitException(int exitCode, Throwable cause) {
		super(cause);
		this.exitCode = exitCode;
	}

	public int getExitCode() {
		return exitCode;
	}
}

File: src/main/kotlin/io/github/saneea/dvh/AppExitException.kt
package io.github.saneea.dvh

class AppExitException @JvmOverloads constructor(val exitCode: Int, cause: Throwable? = null) :
    RuntimeException(cause) {

    interface ExitCode {
        companion object {
            const val OK = 0
            const val ERROR = 1
        }
    }

    companion object {
        /**
         *
         */
        private const val serialVersionUID = 1L
    }
}